src/main/java/org/mariadb/jdbc/internal/com/read/resultset/rowprotocol/TextRowProtocol.java

```diff
@@ -234,8 +234,9 @@ public String getInternalString(ColumnDefinition columnInfo, Calendar cal, TimeZ
     }
 
     if (maxFieldSize > 0) {
-      return new String(buf, pos, Math.min(maxFieldSize * 3, length), StandardCharsets.UTF_8)
-          .substring(0, Math.min(maxFieldSize, length));
+    	String value = new String(buf, pos, Math.min(maxFieldSize * 3, length), StandardCharsets.UTF_8);
+    	int maxLength = Math.min(maxFieldSize, length);
+    	return  value.substring(0, Math.min(maxLength, value.length()));
```

Comment:
```
int maxLength = Math.min(maxFieldSize, length);
return value.substring(0, Math.min(maxLength, value.length()));
```
can be simplified to 

```
if (maxFieldSize < value.length()) {
   return value.substring(0, maxFieldSize);
}
return value;
```
value.length will always be smaller or equal than length.

Reply:
Correct. Changed it to your suggestion.